base/uk.ac.stfc.isis.ibex.ui.devicescreens.tests/src/uk/ac/stfc/isis/ibex/ui/devicescreens/tests/DeviceScreensDescriptionViewModelTest.java

```diff
@@ -134,55 +135,97 @@ public void constructor_produces_deep_copy_of_description() {
     @Test
     public void changing_the_selected_screen_updates_name_key_etc() {
         // Act
-        viewModel.setSelectedScreen(1);
+        List<DeviceDescriptionWrapper> first = viewModel.getScreens().subList(1, 2);
+        viewModel.setSelectedScreens(first);
         
         // Assert
-        assertEquals(deviceName + "2", viewModel.getCurrentName());
+        assertEquals(deviceName + "2", viewModel.getName());
         assertEquals(opiName1, viewModel.getCurrentKey());
-        assertEquals(opiDescription1, viewModel.getCurrentDescription());
-        assertEquals(deviceName + "2", viewModel.getSelectedScreen().getName());
+        assertEquals(opiDescription1, viewModel.getDescription());
     }
 
     @Test
     public void clearing_the_selected_screen_unsets_name_key_etc() {
         // Act
-        viewModel.setSelectedScreen(-1);
+        viewModel.setSelectedScreens(null);
 
         // Assert
-        assertEquals("", viewModel.getCurrentName());
+        assertEquals("", viewModel.getName());
         assertEquals("", viewModel.getCurrentKey());
-        assertEquals("", viewModel.getCurrentDescription());
-        assertEquals(null, viewModel.getSelectedScreen());
+        assertEquals("", viewModel.getDescription());
+    }
+
+    @Test
+    public void setting_multiple_screens_unsets_name_key_etc() {
+        // Act
+        List<DeviceDescriptionWrapper> screens = viewModel.getScreens().subList(0, 2);
+        viewModel.setSelectedScreens(screens);
+
+        // Assert
+        assertEquals("", viewModel.getName());
+        assertEquals("", viewModel.getCurrentKey());
+        assertEquals("", viewModel.getDescription());
+    }
+
+    @Test
+    public void setting_multiple_screens_disables_controls() {
+        // Act
+        List<DeviceDescriptionWrapper> screens = viewModel.getScreens().subList(0, 2);
+        viewModel.setSelectedScreens(screens);
+
+        // Assert
+        assertEquals(false, viewModel.getEnabled());
+    }
+
+    @Test
+    public void setting_multiple_screens_causes_null_target() {
+        // Act
+        List<DeviceDescriptionWrapper> screens = viewModel.getScreens().subList(0, 2);
+        viewModel.setSelectedScreens(screens);
+
+        // Assert
+        assertEquals(null, viewModel.getTargetScreen());
+    }
+
+    @Test
+    public void setting_singular_screen_sets_target() {
+        // Act
+        List<DeviceDescriptionWrapper> first = viewModel.getScreens().subList(0, 1);
+        viewModel.setSelectedScreens(first);
+
+        // Assert
+        assertEquals(first.get(0), viewModel.getTargetScreen());
     }
 
     @Test
     public void changing_the_name_propagates() {
```

Comment:
More poor test naming. I know you haven't touched these but could I appeal to your philanthropic nature to improve all the tests in the class?